pandas/tests/groupby/test_groupby.py

```diff
@@ -1397,7 +1397,7 @@ def f(group):
     df.groupby('a', sort=False, group_keys=False).apply(f)
     # we expect 2 zeros because we call ``f`` once to see if a faster route
     # can be used.
-    expected_names = [0, 0, 1, 2]
+    expected_names = [0, 1, 2]
```

Comment:
@jreback This is the only test I am modifying in this PR. Essentially it tracks how often the function is applied to a given row/group by appending the groups name to a list. 
Previous behavior: Apply it twice to the first row. The first apply is done to check for `fast_apply` compatibility. The second apply is due to looping again from 0 to 2. 
In my change I propose to perform the initial `fast_apply` compatibility check _within_ the loop to save one iteration. 

Is this what you had in mind or what scenario would you like me to test?